File: src/main/java/org/example/bankramenserver/domain/recurring/scheduler/RecurringPaymentReminderScheduler.java
package org.example.bankramenserver.domain.recurring.scheduler;

import lombok.RequiredArgsConstructor;
import org.example.bankramenserver.domain.recurring.service.SendRecurringPaymentReminderService;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class RecurringPaymentReminderScheduler {

    private final SendRecurringPaymentReminderService sendRecurringPaymentReminderService;

    @Scheduled(cron = "0 0 9 * * *", zone = "Asia/Seoul")
    public void sendReminder() {
        sendRecurringPaymentReminderService.execute();
    }
}

File: src/main/kotlin/org/example/bankramenserver/domain/recurring/scheduler/RecurringPaymentReminderScheduler.kt
package org.example.bankramenserver.domain.recurring.scheduler

import org.example.bankramenserver.domain.recurring.service.SendRecurringPaymentReminderService
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component

@Component
class RecurringPaymentReminderScheduler(private val sendRecurringPaymentReminderService: SendRecurringPaymentReminderService) {
    @Scheduled(cron = "0 0 9 * * *", zone = "Asia/Seoul")
    fun sendReminder() = sendRecurringPaymentReminderService.execute()
}
